estate/models/estate_property.py

```diff
@@ -0,0 +1,84 @@
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, fields, models
+
+from odoo.exceptions import UserError
+
+
+class EstateProperty(models.Model):
+    _name = "estate.property"
+    _description = "Real Estate Property"
+    
+    name = fields.Char("Property Name", required=True)
+    description = fields.Text("Description")
+    postcode = fields.Char("Postcode")
+    date_availability = fields.Date("Available From", copy=False, default=fields.Date.today() + relativedelta(months=3))
+    expected_price = fields.Float("Expected Price", required=True)
+    selling_price = fields.Float("Selling Price", readonly=True, copy=False)
+    bedrooms = fields.Integer("Bedrooms", default=2)
+    living_area = fields.Integer("Living Area (sqm)")
+    facades = fields.Integer("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
+    garden_area = fields.Integer("Garden Area (sqm)")
+    active = fields.Boolean(default=True)
+    garden_orientation = fields.Selection( 
+        string="Garden Orientation",
+        selection=[
+            ('north', 'North'), 
+            ('south', 'South'),
+            ('east', 'East'),
+            ('west', 'West'),
+        ],
+    )
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled'),
+        ],
+        required=True, copy=False, default='new',
+    )
+    property_type_id = fields.Many2one('estate.property.type', string="Property Type")
+    user_id = fields.Many2one('res.users', string="Salesman", default=lambda self: self.env.user)
+    partner_id = fields.Many2one('res.partner', string="Buyer", copy=False)
+    tag_ids=fields.Many2many('estate.property.tag', string="Tags")
+    offer_ids=fields.One2many('estate.property.offer', 'property_id', string='Offers')
+    total_area = fields.Integer(string='Total Area (sqm)', compute='_compute_total_area')
+    best_price = fields.Float(string='Best Offer', compute='_compute_best_offer')
+
+    @api.depends('living_area','garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area+record.garden_area
```

Comment:
```suggestion
    def _compute_total_area(self):
        for record in self:
            record.total_area = record.living_area+record.garden_area
```
1.is loop really required?
2. what are decorators and mention some decorators that we use in odoo

Reply:
Not necessarily. We use loop to iterate over records, but odoo will automatically iterate over all records behind the scene. So we can avoid using loop here to iterate over the records.

Decorators in Odoo are special functions (prefixed with @) that modify or extend the behavior of a method without changing its code. 
Decorators that we use in odoo: 
- @api.depends (For Compute Fields)
- @api.constrains (For Validation)
- @api.onchange (For UI Changes Without Saving)